File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BotActiveThemeModel.java
package kore.botssdk.models;

import android.annotation.SuppressLint;

@SuppressLint("UnknownNullness")
public class BotActiveThemeModel
{
    String _id;
    String streamId;
    boolean activeTheme;
    String createdBy;
    String createdOn;
    boolean defaultTheme;
    String lastModifiedBy;
    String lastModifiedOn;
    String refId;
    String state;
    String themeName;
    BrandingBotMessagesModel botMessage;
    BrandingBotMessagesModel userMessage;
    BrandingButtonsModel buttons;
    BrandingGeneralAttributesModel generalAttributes;
    BrandingWidgetBodyModel widgetBody;
    BrandingWidgetBodyModel widgetFooter;
    BrandingWidgetBodyModel widgetHeader;

    public BrandingWidgetBodyModel getWidgetFooter() {
        return widgetFooter;
    }

    public BrandingWidgetBodyModel getWidgetHeader() {
        return widgetHeader;
    }

    public BrandingWidgetBodyModel getWidgetBody() {
        return widgetBody;
    }

    public BrandingBotMessagesModel getUserMessage() {
        return userMessage;
    }

    public BrandingGeneralAttributesModel getGeneralAttributes() {
        return generalAttributes;
    }

    public BrandingButtonsModel getButtons() {
        return buttons;
    }

    public BrandingBotMessagesModel getBotMessage() {
        return botMessage;
    }

    public String get_id() {
        return _id;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public String getCreatedOn() {
        return createdOn;
    }

    public String getLastModifiedBy() {
        return lastModifiedBy;
    }

    public String getLastModifiedOn() {
        return lastModifiedOn;
    }

    public String getRefId() {
        return refId;
    }

    public String getState() {
        return state;
    }

    public String getStreamId() {
        return streamId;
    }

    public String getThemeName() {
        return themeName;
    }

    public boolean isActiveTheme() {
        return activeTheme;
    }

    public boolean isDefaultTheme() {
        return defaultTheme;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BotActiveThemeModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BotActiveThemeModel(
    @SerializedName("_id")
    val id: String = "",
    @SerializedName("streamId")
    val streamId: String = "",
    @SerializedName("activeTheme")
    val activeTheme: Boolean,
    @SerializedName("createdBy")
    val createdBy: String = "",
    @SerializedName("createdOn")
    val createdOn: String = "",
    @SerializedName("defaultTheme")
    val defaultTheme: Boolean,
    @SerializedName("lastModifiedBy")
    val lastModifiedBy: String = "",
    @SerializedName("lastModifiedOn")
    val lastModifiedOn: String = "",
    @SerializedName("refId")
    val refId: String = "",
    @SerializedName("state")
    val state: String = "",
    @SerializedName("themeName")
    val themeName: String = "",
    @SerializedName("v3")
    val brandingModel: BotBrandingModel? = null,
    @SerializedName("botMessage")
    var botMessage: BrandingBotMessagesModel? = null,
    @SerializedName("userMessage")
    var userMessage: BrandingBotMessagesModel? = null,
    @SerializedName("buttons")
    var buttons: BrandingButtonsModel? = null,
    @SerializedName("generalAttributes")
    var generalAttributes: BrandingGeneralAttributesModel? = null,
    @SerializedName("widgetBody")
    var widgetBody: BrandingWidgetBodyModel? = null,
    @SerializedName("widgetFooter")
    var widgetFooter: BrandingWidgetBodyModel? = null,
    @SerializedName("widgetHeader")
    var widgetHeader: BrandingWidgetBodyModel? = null
)

fun BotActiveThemeModel.updateWithV3Model(configModel: BotBrandingModel): BotActiveThemeModel {
    return this.copy(
        brandingModel = brandingModel?.updateWith(configModel),
    )
}